tests/manage/mcg/conftest.py

```diff
@@ -104,6 +108,176 @@ def bucket_cleanup():
     return _create_buckets
 
 
+@pytest.fixture()
+def cloud_uls_factory(request, cld_mgr):
+    """
+        Create a Underlying Storage factory.
+        Calling this fixture creates a new underlying storage(s).
+
+        Args:
+            cld_mgr (CloudManager): Cloud Manager object containing all connections to clouds
+    """
+    created_uls = {
+        'aws': set(),
+        'google': set(),
+        'azure': set(),
+        's3comp': set()
+    }
+
+    ulsMap = {
+        'aws': cld_mgr.aws_client,
+        'google': cld_mgr.google_client,
+        'azure': cld_mgr.azure_client,
+        's3comp': cld_mgr.s3comp_client
+    }
+
+    def _create_uls(uls_dict):
+        """
+        Creates and deletes all underlying storage that were created as part of the test
+
+        Args:
+            uls_dict (dict): Dictionary containing storage provider as key and a list of tuples
+            as value.
+            each tuple contain amount as first parameter and region as second parameter.
+            Example:
+                'aws': [(3,us-west-1),(2,eu-east-2)]
+
+        Returns:
+            dict: A dictionary of cloud names as keys and uls names sets as value.
+        """
+        for cloud, params in uls_dict:
```

Comment:
is that working? Shouldn't you use uls_dict.items() for unpacking this dict?

Reply:
code not yet tested, you are right needs to be .items()